src/Controls/src/Core/Handlers/Items/iOS/ItemsViewController.cs

```diff
@@ -567,7 +567,7 @@ protected virtual void HandleFormsElementMeasureInvalidated(VisualElement formsE
 		internal void UpdateView(object view, DataTemplate viewTemplate, ref UIView uiView, ref VisualElement formsElement)
 		{
 			// Is view set on the ItemsView?
-			if (view is null && viewTemplate is null)
+			if (view is null && (viewTemplate is null || viewTemplate is DataTemplateSelector))
```

Comment:
@jsuarezruiz , 
In this case, the SelectTemplate method itself isn't returning null. Instead, the exception occurs immediately afterward, when the selected viewTemplate is used in CreateContent. This happens because, on the initial render, there's no template available to select, resulting in viewTemplate being null. This leads to an exception when CreateContent is called with an empty or unassigned DataTemplateSelector.

Let me know if further clarification is needed!